reindexer/reindex_request_processor/src/test/scala/uk/ac/wellcome/platform/reindex/processor/services/ReindexWorkerServiceTest.scala

```diff
@@ -0,0 +1,181 @@
+              assertQueueHasSize(dlq, 1)
+              assertDynamoDbHasNoItems[ReindexableRecord](table)
+            }
+          }
+        }
+    }
+  }
+
+  it("updates an existing record with a newer version") {
+    withLocalSqsQueue { queue =>
+      withLocalDynamoDbTable { table =>
+        withReindexWorkerService(table, queue) { _ =>
+          val reindexVersion = 1
+          givenDynamoDbHasItem(
+            SimpleReindexableRecord(id, recordVersion, reindexVersion, data),
+            table)
+
+          val updatedReindexVersion = 2
+          sendMessageInNotification(queue, ReindexRequest(id, updatedReindexVersion))
+
+          eventually {
+            assertDynamoDbOnlyHasItem(
+              SimpleReindexableRecord(
+                id,
+                recordVersion + 1,
+                updatedReindexVersion,
+                data),
+              table)
+          }
+        }
+      }
+    }
+  }
+
+  it("does not update an existing record with an older version") {
+    withLocalSqsQueue { queue =>
+      withLocalDynamoDbTable { table =>
+        withReindexWorkerService(table, queue) { _ =>
+          val originalReindexVersion = 2
+          givenDynamoDbHasItem(
+            SimpleReindexableRecord(
+              id,
+              recordVersion,
+              originalReindexVersion,
+              data),
+            table)
+
+          val updatedReindexVersion = 1
+          sendMessageInNotification(queue, ReindexRequest(id, updatedReindexVersion))
+
+          eventually {
+            assertQueueEmpty(queue)
+            assertDynamoDbOnlyHasItem(
+              SimpleReindexableRecord(
+                id,
+                recordVersion,
+                originalReindexVersion,
+                data),
+              table)
+          }
+        }
+      }
+    }
+  }
+
+  it("fails if saving to dynamo fails") {
+    withLocalSqsQueueAndDlq {
+      case QueuePair(queue, dlq) =>
+        val badTable = Table("table", "index")
+        withReindexWorkerService(badTable, queue) { _ =>
+          val reindexRequest = ReindexRequest("unknownId", 1)
+          sendMessageInNotification(queue, reindexRequest)
+
+          eventually {
+            assertQueueEmpty(queue)
+            assertQueueHasSize(dlq, 1)
+          }
+        }
+    }
+  }
+
+  it("updates a sourcedata record") {
+    withLocalSqsQueue { queue =>
+      withLocalDynamoDbTable { table =>
+        withReindexWorkerService(table, queue) { _ =>
+          case class SourceDataRecord(id: String,
+                                      version: Int,
+                                      reindexVersion: Int,
+                                      reIndexShard: String,
+                                      s3key: String,
+                                      sourceName: String,
+                                      sourceId: String)
+          val sourceDataRecord = SourceDataRecord(
+            id,
+            1,
+            100,
```

Comment:
Numbers without keyword arguments make me twitchy, because it takes a moment of back and forth to establish this is the reindex version. Keyword arguments please?